modules/alloy/src/test/ts/atomic/alien/BoxesTest.ts

```diff
@@ -0,0 +1,57 @@
+import { context, describe, it } from '@ephox/bedrock-client';
+import { assert } from 'chai';
+import fc from 'fast-check';
+
+import * as Boxes from 'ephox/alloy/alien/Boxes';
+
+// TODO: Add more tests.
+describe('atomic.alloy.alien.BoxesTest', () => {
+  context('Boxes.constrain', () => {
+    it('TINY-9226: Constraint equals box', () => {
+      const actual = Boxes.constrain(
+        Boxes.bounds(0, 0, 1000, 500),
+        Boxes.bounds(0, 0, 1000, 500)
+      );
+
+      assert.deepEqual(
+        actual,
+        Boxes.bounds(0, 0, 1000, 500)
+      );
+    });
+
+    it('TINY-9226: Constraint larger than box', () => {
+      const original = Boxes.bounds(100, 100, 800, 300);
+      const constraint = Boxes.bounds(0, 0, 1000, 500);
+      const actual = Boxes.constrain(original, constraint);
+
+      assert.deepEqual(
+        actual,
+        original
+      );
+    });
+
+    context('Properties', () => {
+      const arbBounds: fc.Arbitrary<Boxes.Bounds> = fc.nat(1000).chain((x) =>
```

Comment:
I think you can use the fc.tuple here instead like `fc.tuple(fs.nat(1000), fs.nat(1000), fs.nat(1000), fs.nat(1000))` or something then make a box out of that to avoid the pyramid here.

Reply:
Ah, K. Thanks for the tip.

- [ ] use tuples